src/layers/domain/core/cpm_system_id/tests/test_cpm_system_id_v1.py

```diff
@@ -250,16 +262,47 @@ def test_asid_generation_increment():
         assert generator.latest_id == str(expected_latest)
 
         # Update repository with new number
-        repository.create_or_update(key_name="ASIDNUMBER", new_number=generator.latest)
+        repository.create_or_update(
+            key_name=CpmSystemIdType.ASIDNUMBER, new_number=generator.latest
+        )
 
         # Fetch the updated ID and check if it was correctly incremented
-        new_id = repository.read(key_name="ASIDNUMBER")
+        new_id = repository.read(key_name=CpmSystemIdType.ASIDNUMBER)
         assert int(new_id["latest"]) == expected_latest
 
         # Update for next iteration
         previous_latest = expected_latest
         current_id = new_id
 
     # Final assertion to check if latest is num_calls greater than start_value
-    final_latest = int(repository.read(key_name="ASIDNUMBER")["latest"])
+    final_latest = int(repository.read(key_name=CpmSystemIdType.ASIDNUMBER)["latest"])
     assert final_latest == start_value + num_calls
+
+
+def test_product_id_generator_format_key():
+    generator = ProductId.create()
+    assert generator.latest_id is not None
+
+
+def test_product_id_generator_validate_key_valid():
+    valid_key = "P.AAA-333"
+    is_valid = ProductId.validate_key(valid_key)
+    assert is_valid
+
+
+@pytest.mark.parametrize(
+    "invalid_key",
+    [
+        "ABC000124",  # Missing 'P.' and hyphen
+        "P-ABC-123",  # Extra hyphen
+        "P.ABC123",  # Missing hyphen
+        "P.ABC-1234",  # Too many digits
+        "P.ABC-12",  # Too few digits
+        "P.AB1-123",  # ODS code contains a digit
```

Comment:
Slightly confusing comments here as there is no ODS code involved and the the sections arent specifically digit or not - they are alphanumeric with a reduced choice (due to confusing characters) so worth updating these :)

Reply:
Good spot, some of these are actually wrong anyway! They use invalid characters so would always fail regardless 🤦‍♀️